parquet-column/src/main/java/org/apache/parquet/io/api/Binary.java

```diff
@@ -71,7 +83,28 @@ public boolean equals(Object obj) {
 
   @Override
   public String toString() {
-    return "Binary{" + length() + " bytes, " + Arrays.toString(getBytes()) + "}";
+    return "Binary{" +
+        length() +
+        (isReused ? " reused": " constant") +
+        " bytes, " +
+        Arrays.toString(getBytesUnsafe())
+        + "}";
+  }
+
+  public Binary copy() {
+    if (isReused) {
+      return Binary.fromConstantByteArray(getBytes());
+    } else {
+      return this;
+    }
+  }
+
+  /**
+   * Signals if backing bytes are owned, and can be modified, by producer of the Binary
+   * @return if backing bytes are held on by producer of the Binary
+   */
+  public boolean isReused() {
+    return isReused;
```

Comment:
I'd like there to be a more clear name than `isReused` but I can't think of one... maybe `isFinal` or `backingBytesReassignable` or something?